sycl/doc/extensions/experimental/sycl_ext_oneapi_virtual_mem.asciidoc

```diff
@@ -0,0 +1,398 @@
+
+|`device get_device() const` |
+Returns the SYCL device associated with the instance of `physical_mem`.
+
+|`size_t size() const` |
+Returns the size of the corresponding physical memory in bytes.
+
+|============================
+
+Virtual memory address ranges are mapped to the a `physical_mem` through the
+`map` member functions, where the access mode can also be specified.
+To further get or set the access mode of a mapped virtual address range, the
+user does not need to know the associated `physical_mem` and can just call the
+following free functions.
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+void set_access_mode(const void *ptr, size_t numBytes, address_access_mode mode, const context &syclContext);
+
+address_access_mode get_access_mode(const void *ptr, size_t numBytes, const context &syclContext);
+
+void unmap(const void *ptr, size_t numBytes, const context &syclContext);
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+[frame="topbot",options="header,footer"]
+|=====================
+|Function |Description
+
+|`void set_access_mode(const void *ptr, size_t numBytes, address_access_mode mode, const context &syclContext)` |
+Changes the access mode of a virtual memory range specified by `ptr` and
+`numBytes`.
+
+If `mode` is `address_access_mode::read` or `address_access_mode::read_write`
+`ptr` pointer is accessible after the call as read-only or read-write
+respectively. Otherwise, it is considered inaccessible and accessing it will result
+in undefined behavior.
+
+The virtual memory range specified by `ptr` and `numBytes` can be a sub-range of
+virtual memory ranges previously mapped to `physical_mem`. `ptr`
+must be aligned to the minimum memory granularity of the device associated with
+the `physical_mem` the range is mapped to and `numBytes` must be a multiple of
+the minimum memory granularity of the device associated with the `physical_mem`
+the range is mapped to.
+
+Writing to any address in the virtual memory range with access mode set to
+`address_access_mode::read` results in undefined behavior.
+
+An accessible pointer behaves the same as a pointer to device USM memory and can
+be used in place of a device USM pointer in any interface accepting one.
+
+|`address_access_mode get_access_mode(const void *ptr, size_t numBytes, const context &syclContext)` |
+Returns the access mode of the virtual memory range specified by `ptr` and
+`numBytes`.
+
+The virtual memory range specified by `ptr` and `numBytes` can be a sub-range of
+virtual memory ranges previously mapped to `physical_mem`. `ptr`
+must be aligned to the minimum memory granularity of the device associated with
+the `physical_mem` the range is mapped to and `numBytes` must be a multiple of
+the minimum memory granularity of the device associated with the `physical_mem`
+the range is mapped to.
+
+|`void unmap(const void *ptr, size_t numBytes, const device &syclDevice, const context &syclContext)` |
```

Comment:
is my understanding correct that unmap will change get_access_mode return value to "none"? or it is prohibited to call that functions on unmapped ptr?

"Virtual memory address ranges are mapped to the a `physical_mem` through the
`map` member functions, where the access mode can also be specified.
_**To further get or set the access mode of a mapped virtual address range,**_ the
user does not need to know the associated `physical_mem` and can just call the
following free functions."

Reply:
I have clarified `set_access_mode` and `get_access_mode` to specify that they only work with mapped memory ranges. Any region that has been unmapped cannot be used in these.